src/post_process/m_data_output.f90

```diff
@@ -973,9 +1106,14 @@ subroutine s_finalize_data_output_module() ! -------------------------
         ! that were written to the formatted database file(s). Note that the
         ! root variable is only deallocated in the case of a 1D computation.
         deallocate (q_sf)
+        if (coarsen_silo) deallocate (coarse_x_q_sf, coarse_xy_q_sf, coarse_xyz_q_sf)
         if (n == 0) deallocate (q_root_sf)
         if (grid_geometry == 3) then
-            deallocate (cyl_q_sf)
+            if (coarsen_silo) then
```

Comment:
@wilfonba removed this feature intentionally (it is fairly useless). Please do not add it back.